contrib/win32/openssh/telemetry.c

```diff
@@ -0,0 +1,177 @@
+/* 
+this file defines functions to collect Microsoft Telemetry,
+which will only be sent for Windows In-Box releases. 
+GitHub releases will not send any Telemetry. 
+*/
+
+#include <stdio.h>
+#include <Objbase.h>
+
+#include "telemetry.h"
+#include "telemetryInternal.h"
+
+// {0d986661-0dd7-561a-b15b-fcc1cd46d2bb}
+TRACELOGGING_DEFINE_PROVIDER(
+    g_hProvider1,
+    "Microsoft.Windows.Win32OpenSSH",
+    (0x0d986661, 0x0dd7, 0x561a, 0xb1, 0x5b, 0xfc, 0xc1, 0xcd, 0x46, 0xd2, 0xbb),
+    TraceLoggingOptionMicrosoftTelemetry());
+
+void send_auth_telemetry(const int status, const char* auth_type)
+{
+    TraceLoggingRegister(g_hProvider1);
```

Comment:
I thought about this, as well. My reasoning was to always ensure the provider gets properly unregistered, since I can't guarantee that ssh/sshd will make it to the last telemetry call, if I were to only do it once. What would be the best place to unregister the provider (if necessary)?